examples/demo-app/src/cloud-providers/carto/carto-provider.js

```diff
@@ -0,0 +1,424 @@
+   * @param {*} location
+   */
+  getAccessTokenFromLocation(location) {
+    return;
+  }
+
+  async downloadMap(queryParams) {
+    try {
+      const {owner: username, mapId, privateMap} = queryParams;
+
+      if (!username || !mapId) {
+        return;
+      }
+
+      let visualization;
+
+      if (privateMap.trim().toLowerCase() === 'true') {
+        await this._carto.login();
+        const currentUsername = this.getUserName();
+        if (currentUsername && currentUsername === username) {
+          const cs = await this._carto.getCustomStorage();
+          visualization = await cs.getVisualization(mapId);
+        }
+      } else {
+        visualization = await this._carto.PublicStorageReader.getVisualization(username, mapId);
+      }
+
+      if (!visualization) {
+        throw new Error(`Can't find map with ID: ${mapId}`);
+      }
+
+      // These are the options required for the action. For now, all datasets that come from CARTO are CSV
+      const datasets = visualization.datasets.map(dataset => {
+        const datasetId = dataset.name;
+
+        return {
+          info: {
+            id: datasetId,
+            label: datasetId,
+            description: dataset.description,
+            dataUrl: '',
+            configUrl: '',
+            panelDisabled: true
+          },
+          data: dataset.file
+        };
+      });
+
+      // const datasets = visualization.datasets.map(dataset => dataset.file);
+
+      this.currentMap = visualization.vis;
+
+      return {
+        map: {
+          datasets,
+          config: visualization.vis.config,
+          info: {title: visualization.vis.name, description: visualization.vis.description}
+        },
+        format: 'csv'
+      };
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  async listMaps() {
+    // TODO: Implement pagination using {type='all', pageOffset=0, pageSize=-1}
+    try {
+      await this._carto.login();
+      const username = this.getUserName();
+      const cs = await this._carto.getCustomStorage();
+
+      const visualizations = await cs.getVisualizations();
+      let formattedVis = [];
+
+      // Format visualization object
+      for (const vis of visualizations) {
+        formattedVis.push({
+          id: vis.id,
+          title: vis.name,
+          description: vis.description,
+          privateMap: vis.isPrivate,
+          thumbnail: vis.thumbnail === 'undefined' ? null : vis.thumbnail,
+          lastModification: new Date(Date.parse(vis.lastModified)),
+          loadParams: {
+            owner: username,
+            mapId: vis.id,
+            privateMap: vis.isPrivate.toString()
+          }
+        });
+      }
+
+      formattedVis = formattedVis.sort((a, b) => b.lastModification - a.lastModification);
+
+      return formattedVis;
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  getMapPermalink(mapLink, fullURL = true) {
```

Comment:
no longer called by UI, can make it private

Reply:
👍 Not needed anymore, deleted